lib/activerecord-bitemporal/visualizer.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module ActiveRecord::Bitemporal
+  module Visualizer
+    # Figure is a two-dimensional array holding plotted lines and columns
+    class Figure < Array
+      def print(str, line: 0, column: 0)
+        self[line] ||= []
+        str.each_char.with_index(column) do |c, i|
+          self[line][i] = c
+        end
+      end
+
+      def to_s
+        map { |l| l&.map { |c| c || ' ' }&.join }.join("\n")
+      end
+    end
+
+    module_function
+
+    def visualize(record, height: 10, width: 40, highlight: true)
+      histories = record.class.ignore_valid_datetime.within_deleted.bitemporal_for(record)
```

Comment:
You can use `ignore_bitemporal_datetime` instead of `ignore_valid_datetime.within_deleted`.

```suggestion
      histories = record.class.ignore_bitemporal_datetime.bitemporal_for(record)
```